core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -54,11 +59,84 @@ object ClientQuotaManagerConfig {
   val DefaultQuotaWindowSizeSeconds = 1
   // Purge sensors after 1 hour of inactivity
   val InactiveSensorExpirationTimeSeconds  = 3600
+
+  val UnlimitedQuota = Quota.upperBound(Long.MaxValue)
+  val DefaultClientIdQuotaId = QuotaId.quotaId(None, Some(ConfigEntityName.Default))
+  val DefaultUserQuotaId = QuotaId.quotaId(Some(ConfigEntityName.Default), None)
+  val DefaultUserClientIdQuotaId = QuotaId.quotaId(Some(ConfigEntityName.Default), Some(ConfigEntityName.Default))
+}
+
+object QuotaTypes {
+  val NoQuotas = 0
+  val ClientIdQuotaEnabled = 1
+  val UserQuotaEnabled = 2
+  val UserClientIdQuotaEnabled = 4
+}
+
+object QuotaId {
+
+  val Separator = ':'
```

Comment:
Will this cause issues in jmx name? It seems that : is used to separate types in jmx? Similarly, are %, <, > allowed in jmx name? Is < > allows in ZK path?